Gabi/atqr 7 page hora da verdade

GabrielaRsanches opened the pull request.

Commit:
feat: changed static function updateStatus to public to use it in timeForTheTruth.ts

Commit:
feat: made error treatment better

Commit:
feat: implemented timeForTheTruth.html and   timeForTheTruth.ts

Commit:
feat: updated imports from Meeta and finished logic to change challenge status

Commit:
feat: created pug file for timeForTheTruth

albnunes commented on apps/atqr/web/src/pages/timeForTheTruth/timeForThruth.pug:
```diff
@@ -0,0 +1,38 @@
+doctype html
```
Your pug file should be running.  The cdns and ts file imports are missing. Do this and run with parcel

albnunes commented on apps/atqr/web/src/pages/timeForTheTruth/timeForTheTruth.html:
```diff
@@ -0,0 +1,111 @@
+<!DOCTYPE html>
```
we don't use html files. remove this file

albnunes reviewed (changes requested):
resolve the comments

Commit:
pull request changes: deleted html files, imported cdns and ts files on timeFortheTruth.pug

Commit:
finished reviewing task

yfernandes commented on apps/atqr/api/src/app/challenge.controller.ts:
```diff
@@ -157,7 +157,7 @@ export class ChallengeController {
   }
 
-  private async updateStatus(id: Guid, status: ChallengeStatus): Promise<void> {
+   async updateStatus(id: Guid, status: ChallengeStatus): Promise<void> {
```
What was the reasoning to remove the private keyword?
If your intention was to make the updateStatus method publicly available you failed to include the required
decorator to make it a endpoint. Remember that you are in the context of a controller.
Howerver, this method was intend to be private and to only contain the logic pertaing to statuses updates.
The endpoint that would be called, and later call this method is the endpoint updateChallenge (line 127).

Also, beware of formatting and linting before commiting and making PR. There's an extra space before async.
We have automated tools that check for that PRs and Commits that dont follow the style guide will no longer
be accepted by me.

GabrielaRsanches replied:
I did it because im using the updateStaus function ( of challenge.controlller ) to use in timeForTheThruth.ts on its buttons

yfernandes commented on apps/atqr/web/src/pages/challenge/challenge.ts:
```diff
@@ -1,45 +1,44 @@
 
-
-
 const getChallenge = async () => {
```
In order to keep a clear separation of concerns please refrain from creating and manipulating axios instances inside page specific .ts files.
An api.ts file (located in the services folder) was created in order to contain and interface with the backend.

GabrielaRsanches replied:
so I have to put all axios instances/logic in the api.ts file ? like, all of the logic to get things from web ?

yfernandes replied:
In order to keep our code DRY yes. All outgoing API requests should be made from the services/api.ts file. This ensures that we have a clear separation of concerns, deal with API calls in one place only, and handle all API calls equally. This is also better for maintenance.

From the point of view of the page, you won't have to deal with API call just business rules and HTML handling, which will make the file much less cluttered.

yfernandes replied:
Re-Resolve This one

yfernandes commented on apps/atqr/web/src/pages/timeForTheTruth/timeForTheTruth.ts:
```diff
@@ -0,0 +1,27 @@
+import { ChallengeStatus, Email } from '@atqr/domain';
```
Unused imports should be deleted

yfernandes commented on apps/atqr/web/src/pages/timeForTheTruth/timeForTheTruth.ts:
```diff
@@ -0,0 +1,27 @@
+import { ChallengeController } from '../../../../api/src/app/challenge.controller';
+import { challenge } from '../challenge/challenge';
+
+export class TimeForTheTruth {
```
This class is never used

yfernandes commented on apps/atqr/web/src/pages/timeForTheTruth/timeForTheTruth.ts:
```diff
@@ -0,0 +1,27 @@
+import { challenge } from '../challenge/challenge';
+
+export class TimeForTheTruth {
+  constructor(private challengeController: ChallengeController) {}
```
You cannot use a backend (node , nestjs) class as parameter for a frontend class (web, html)

yfernandes reviewed (changes requested):
Many changes are required, try to keep your intentions clear and you code concise.

Commit:
resolve: did changes that ygo pointed on PR

yfernandes commented:
Hey @GabrielaRsanches, I was taking a deeper look at this PR late at night yesterday and couldn't get it to work, can you post a print screen or share how you got to your final result?

Commit:
refactor(apiclient.ts): file name is now apiClient, not api

Commit:
refactor(timeforthetruth.ts and js): congrats and reprove buttons use clientApi to change status

uses getUrlId() from urlDetails

BREAKING CHANGE: web is now quering strings from browser urls

Commit:
refactor(urldetails.ts and js on web): creation of urlDetails to implement getUrlId()

gets url from browser and extraxts the id of it. It is on a separate file to be reused and the file
to hold funcions linked to url

yfernandes commented on apps/atqr/web/src/pages/timeForTheTruth/timeForTheTruth.js:
```diff
@@ -0,0 +1,94 @@
+"use strict";
```
.JS files should never be commited in a Typescript project. Delete this and the UrlDetails.js file

yfernandes commented on apps/atqr/api/src/app/challenge.controller.ts:
```diff
@@ -134,20 +134,7 @@ export class ChallengeController {
 
```
When performing a rebase from another branch be VERY carefull with what code is being kept
and what code is being deleted. A piecer of this method was almost lost!

yfernandes commented on apps/atqr/web/src/pages/acceptChallenge/acceptChallenge.ts:
```diff
@@ -4,7 +4,7 @@ const axiosInstance = axios.create({
   baseURL: 'https://jsonplaceholder.typicode.com/',
```
This file should be deleted!

yfernandes commented on apps/atqr/web/src/pages/timeForTheTruth/timeForTheTruth.ts:
```diff
@@ -0,0 +1,41 @@
+import { atqrApi } from '../../services/apiClient';
+
+const idUrl = getUrlId.toString();
+const urlId = idUrl;
```
Avoid using unnecessary intermediary variables

yfernandes commented on apps/atqr/web/src/pages/timeForTheTruth/timeForTheTruth.ts:
```diff
@@ -0,0 +1,41 @@
+    return challengeStatusUpdated;
+  }
+}
+congratsButton();
```
This function shoudn't be called here. It should be linked to the form button and be executed on the user's click

yfernandes commented on apps/atqr/web/src/pages/timeForTheTruth/timeForTheTruth.ts:
```diff
@@ -0,0 +1,41 @@
+    return challengeStatusUpdated;
+  }
+}
+reproveButton();
```
Same as above

yfernandes commented on apps/atqr/web/src/services/apiClient.ts:
```diff
@@ -1,6 +1,6 @@
+import { ChallengeStatus } from '@atqr/domain';
 import { Guid } from '@tokilabs/lang';
-import axios from 'axios';
+import axios, { AxiosError } from 'axios';
```
Unused imports should be removed.

yfernandes commented on apps/atqr/web/src/services/urlDetails.ts:
```diff
@@ -0,0 +1,10 @@
+const urlSearchParams = new URLSearchParams(url);
+export async function getUrlId() {
+  try {
+    await urlSearchParams.getAll('id');
```
Unnecessary Async/Await

yfernandes commented on apps/atqr/web/src/services/urlDetails.ts:
```diff
@@ -0,0 +1,10 @@
+const url = window.location.pathname;
+const urlSearchParams = new URLSearchParams(url);
+export async function getUrlId() {
```
Try to refactor this funtion to be a little more generic to allow for more flexibility and use cases

yfernandes commented on apps/atqr/web/src/services/apiClient.ts:
```diff
@@ -73,6 +73,16 @@ export const atqrApi = {
         }
       },
     },
+    getOne: async (id: string) => {
```
Try creating a mock function of this one to help developing your page.

yfernandes reviewed (changes requested):
Looks like the Time for the truth page is still unfinished, but i see your progress. Good job! =D

Keep me posted, and share a few screenshots of your page as soon as you can.

yfernandes reviewed (dismissed):
Looks like the Time for the truth page is still unfinished, but i see your progress. Good job! =D

Keep me posted, and share a few screenshots of your page as soon as you can.

Commit:
refactor(wip): wIP

Commit:
refactor(deadline.ts, deadline.pug and deadlineindex): timeForTheTruth is called dealine. Page is ok

timeForTheTruth is now called dealine. File ts is now using generic function updateUi from services
to display the information. Pug is ready and appling cdns imports

Commit:
refactor(updateui.ts): generic function to get elements with querySelectorAll and display

Commit:
refactor(urldetails.ts): function is now generic

Commit:
refactor(apiclient.ts and interfaces.ts): interfaces.ts holds interfaces to be used in frontend

ApiClient now uses interfaces that represent the actual classes that have to implement them in the
future

BREAKING CHANGE: frontend now uses interfaces instead of the actual classes

Commit:
refactor(challenge, challengeindex, createchallenge, lastplayerssession and index (all ".pug")): pag

pug pages are compiling and running. Challenge.pug renders challenge resume, challengeIndex has its
content.

Commit:
refactor(files changed that have nothing to do with the task): files changed

dont know why some changes happend but im commiting because I fear making a mass otherwise

Commit:
style(js files): js file were deleted

albnunes committed:
refactor(web): pug files are running

albnunes reviewed (approved):
it's ok

albnunes committed:
refactor(web): path to components is fixed
